services/metalv1/client.go

```diff
@@ -354,31 +363,18 @@ func parameterToJson(obj interface{}) (string, error) {
 // callAPI do the request.
 func (c *APIClient) callAPI(request *http.Request) (*http.Response, error) {
 	if c.cfg.Debug {
-		authToken, hasAuth := request.Header["X-Auth-Token"]
```

Comment:
The common template doesn't obfuscate the `X-Auth-Token` header, so if we want to keep this code in place, we would either need to restore the `client.mustache` override template for `metalv1` or add obfuscation of this header to the shared template even though other services don't use this header.

IMO it's OK to lose this header obfuscation.  The `Debug` flag is just one way to enable wire logging for this client; in the Terraform provider we use Hashicorp's logging transport, which doesn't obfuscate anything.